catroid/src/main/java/org/catrobat/catroid/common/BroadcastSequenceMap.java

```diff
@@ -28,43 +28,43 @@
 import java.util.HashMap;
 import java.util.Map;
 
-public final class BroadcastSequenceMap {
-	private static Map<String, HashMap<String, ArrayList<SequenceAction>>> broadcastSequenceMap = new HashMap<>();
+public class BroadcastSequenceMap {
+	protected Map<String, Map<String, ArrayList<SequenceAction>>> broadcastSequenceMap = new HashMap<>();
 
-	private BroadcastSequenceMap() {
-		throw new AssertionError();
+	public boolean containsKey(String key, String sceneName) {
+		return broadcastSequenceMap.containsKey(sceneName) && broadcastSequenceMap.get(sceneName).containsKey(key);
 	}
 
-	public static boolean containsKey(String key, String sceneName) {
-		if (!BroadcastSequenceMap.broadcastSequenceMap.containsKey(sceneName)) {
-			return false;
+	public ArrayList<SequenceAction> get(String key, String sceneName) {
+		Map<String, ArrayList<SequenceAction>> map = broadcastSequenceMap.get(sceneName);
+		if (map == null) {
+			map = new HashMap<>();
+			broadcastSequenceMap.put(sceneName, map);
+		}
+		if (!map.containsKey(key)) {
+			map.put(key, new ArrayList<SequenceAction>());
 		}
-		return BroadcastSequenceMap.broadcastSequenceMap.get(sceneName).containsKey(key);
+		return map.get(key);
 	}
 
-	public static ArrayList<SequenceAction> get(String key, String sceneName) {
-		if (!BroadcastSequenceMap.broadcastSequenceMap.containsKey(sceneName)) {
-			return null;
-		}
-		return BroadcastSequenceMap.broadcastSequenceMap.get(sceneName).get(key);
+	public boolean add(String sceneName, String key, SequenceAction value) {
```

Comment:
why rename to add? up until now BroadcastSequenceMap although not extending Map kind of behaved like what one would expect from something called *Map.